File: app/src/main/java/com/isp/app/ui/programs/adapter/ProgramsAdapter.java
package com.isp.app.ui.programs.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.isp.app.R;
import com.isp.app.data.models.Program;
import java.util.List;

public class ProgramsAdapter extends RecyclerView.Adapter<ProgramsAdapter.ProgramViewHolder> {

    private final List<Program> programs;

    public ProgramsAdapter(List<Program> programs) {
        this.programs = programs;
    }

    @NonNull
    @Override
    public ProgramViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_program, parent, false);
        return new ProgramViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ProgramViewHolder holder, int position) {
        Program program = programs.get(position);
        holder.programName.setText(program.getName());
    }

    @Override
    public int getItemCount() {
        return programs.size();
    }

    static class ProgramViewHolder extends RecyclerView.ViewHolder {
        TextView programName;

        ProgramViewHolder(@NonNull View itemView) {
            super(itemView);
            programName = itemView.findViewById(R.id.programName);
        }
    }
} 

File: app/src/main/java/com/college/app/ui/programs/adapter/ProgramsAdapter.kt
package com.college.app.ui.programs.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.college.app.data.models.Program
import com.college.app.databinding.ItemProgramBinding

class ProgramsAdapter(
    private val onProgramClick: (Program) -> Unit
) : ListAdapter<Program, ProgramsAdapter.ProgramViewHolder>(ProgramDiffCallback()) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ProgramViewHolder {
        val binding = ItemProgramBinding.inflate(
            LayoutInflater.from(parent.context),
            parent,
            false
        )
        return ProgramViewHolder(binding, onProgramClick)
    }

    override fun onBindViewHolder(holder: ProgramViewHolder, position: Int) {
        holder.bind(getItem(position))
    }

    class ProgramViewHolder(
        private val binding: ItemProgramBinding,
        private val onProgramClick: (Program) -> Unit
    ) : RecyclerView.ViewHolder(binding.root) {

        fun bind(program: Program) {
            binding.programName.text = program.name
            binding.root.setOnClickListener { onProgramClick(program) }
        }
    }

    private class ProgramDiffCallback : DiffUtil.ItemCallback<Program>() {
        override fun areItemsTheSame(oldItem: Program, newItem: Program): Boolean {
            return oldItem.id == newItem.id
        }

        override fun areContentsTheSame(oldItem: Program, newItem: Program): Boolean {
            return oldItem == newItem
        }
    }
} 